python_modules/libraries/dagster-airlift/dagster_airlift/core/airflow_instance.py

```diff
@@ -75,24 +78,31 @@ def get_api_url(self) -> str:
         return f"{self.auth_backend.get_webserver_url()}/api/v1"
 
     def list_dags(self) -> list["DagInfo"]:
-        response = self.auth_backend.get_session().get(
-            f"{self.get_api_url()}/dags", params={"limit": 1000}
-        )
-        if response.status_code == 200:
-            dags = response.json()
-            webserver_url = self.auth_backend.get_webserver_url()
-            return [
-                DagInfo(
-                    webserver_url=webserver_url,
-                    dag_id=dag["dag_id"],
-                    metadata=dag,
-                )
-                for dag in dags["dags"]
-            ]
-        else:
-            raise DagsterError(
-                f"Failed to fetch DAGs. Status code: {response.status_code}, Message: {response.text}"
+        dag_responses = []
+        webserver_url = self.auth_backend.get_webserver_url()
+        while True:
+            prev_len = len(dag_responses)
```

Comment:
nit but keeping track of the previous length seems more complicated than just checking if the latest batch size is 0/less than the limit

Reply:
🤦‍♂️ u right